src/gemm.jl

```diff
@@ -54,53 +56,68 @@ end
+        t = Base.Cartesian.@ntuple $N n -> VectorizationBase._vbroadcast(StaticInt{$W}(), $(T(-Inf)), StaticInt{$RS}())
+        Tropical(VecUnroll(t))
+    end
 end
 
 @inline function Base.promote_rule(::Type{Tropical{T1}}, ::Type{Tropical{T2}}) where {T1<:VecUnroll,T2<:Vec}
     Tropical{promote_rule(T1, T2)}
 end
 
-@inline function VectorizationBase._vzero(::StaticInt{W}, ::Type{T}, ::StaticInt{RS}) where {W,T<:Tropical{FT},RS} where FT
+@inline function VectorizationBase._vzero(::StaticInt{W}, ::Type{T}, ::StaticInt{RS}) where {W,FT,T<:Tropical{FT},RS}
     Tropical(VectorizationBase._vbroadcast(StaticInt{W}(), FT(-Inf), StaticInt{RS}()))
 end
 
-@inline function VectorizationBase.fma(x::Tropical{V}, y::Tropical{V}, z::Tropical{V}) where {V<:VectorizationBase.AbstractSIMD}
-    Tropical(max(content(z), content(x) + content(y)))
+@inline function Base.fma(x::Tropical{V}, y::Tropical{V}, z::Tropical{V}) where {V<:VectorizationBase.AbstractSIMD}
+    Tropical(Base.FastMath.max_fast(content(z), Base.FastMath.add_fast(content(x), content(y))))
+end
+@inline function Base.fma(::StaticInt{N}, y::Tropical{V}, z::Tropical{V}) where {N,V<:VectorizationBase.AbstractSIMD}
+    Base.FastMath.add_fast(Base.FastMath.mul_fast(StaticInt{N}(), y), z)
 end
 
-# is `gep` a shorthand for "get element pointer"?
+# `gep` is a shorthand for "get element pointer"
 @inline VectorizationBase.gep(ptr::Ptr{Tropical{T}}, i) where T = Ptr{Tropical{T}}(VectorizationBase.gep(Ptr{T}(ptr), i))
 
-# TODO: FIX!!!!!!
-@inline function Base.promote(a::Int, b::Tropical{T}) where {T<:Vec}
-    elem = a == 0 ? -Inf : 0.0
-    Tropical(T(elem)), b
+for f ∈ [:(Base.:(*)), :(Base.FastMath.mul_fast)]
+    @eval begin
+        @inline $f(::StaticInt{0}, vx::Tropical{T}) where {T<:AbstractSIMD} = zero(Tropical{T})
+        @inline $f(::StaticInt{1}, vx::Tropical{T}) where {T<:AbstractSIMD} = vx
+        @inline $f(vx::Tropical{T}, ::StaticInt{0}) where {T<:AbstractSIMD} = zero(Tropical{T})
+        @inline $f(vx::Tropical{T}, ::StaticInt{1}) where {T<:AbstractSIMD} = vx
+    end
 end
-
-@inline function Base.promote(a::Int, b::Tropical{T}, c::Tropical{T}) where {T<:Vec}
-    elem = a == 0 ? -Inf : 0.0
-    Tropical(T(elem)), b, c
+for f ∈ [:(Base.:(+)), :(Base.FastMath.add_fast)]
+    @eval begin
+        @inline $f(::StaticInt{0}, vx::Tropical{T}) where {T<:AbstractSIMD} = vx
+        @inline $f(vx::Tropical{T}, ::StaticInt{0}) where {T<:AbstractSIMD} = vx
+    end
 end
-
-@inline function Base.promote(a::Int, b::Tropical{T}) where {T<:VecUnroll}
-    elem = a == 0 ? -Inf : 0.0
-    Tropical(T(elem)), b
+# julia 1.5 patch
+@inline function VectorizationBase.VecUnroll(data::Tuple{T,Vararg{T,N}}) where {N,T<:Tropical}
+    Tropical(VecUnroll(map(content, data)))
 end
 
-@inline function Base.promote(a::Int, b::Tropical{T}, c::Tropical{T}) where {T<:VecUnroll}
-    elem = a == 0 ? -Inf : 0.0
-    Tropical(T(elem)), b, c
+@inline LoopVectorization.vecmemaybe(x::Tropical) = x
```

Comment:
If I comment out `vecmaybe` function on my local host, tests still pass. What is the purpose of this function?